src/app/route/ProductPage/ProductPage.component.js

```diff
@@ -43,27 +48,39 @@ class ProductPage extends Component {
         this.updateBreadcrumbs();
     }
 
-    /**
-     * Get selected configurable product variant
-     * @param {Object} props
-     * @return {Number} variant index
-     */
-    static getVariantIndexFromProps(props) {
-        const { location: { state: locationState } } = props;
+    static getDerivedStateFromProps(props, state) {
+        const { id: stateId } = state;
+        const {
+            product: {
+                variants, configurable_options, id, type_id
+            },
+            location: { search }
+        } = props;
+
+        if (!(configurable_options && variants && id !== stateId)) return null;
+
+        const parameters = Object.entries(convertQueryStringToKeyValuePairs(search))
+            .reduce((acc, [key, value]) => {
+                if (configurable_options.find(({ attribute_code }) => attribute_code === key)) {
+                    return { ...acc, [key]: value };
+                }
+                return acc;
+            }, {});
+
+        const configurableVariantIndex = getVariantIndex(variants, parameters);
+
+        if (Object.keys(parameters).length === Object.keys(configurable_options).length
```

Comment:
Let's agree to use following structure of complex ifs:

```js
if (
    Object.keys(parameters).length === Object.keys(configurable_options).length
    || type_id !== 'configurable'
) {
    return { id, parameters, configurableVariantIndex };
}
```

Reply:
Why not, btw `return { id, parameters, configurableVariantIndex };` always, regarding of this check? Just makes sure it returns undefined or some non integer value from `getVariantIndex` function.